common/src/unifyfs_meta.h

```diff
@@ -102,6 +103,34 @@ typedef struct {
     int gfid;       /* global file id */
 } unifyfs_index_t;
 
+/*
+ * Hash a file path to an unsigned integer using MD5
+ * @param path absolute file path
+ * @return hash value
+ */
+static inline
+unsigned int compute_path_md5(const char* path)
+{
+    unsigned char digested[16] = {0};
+    unsigned long len = strlen(path);
+
+    MD5((const unsigned char*) path, len, digested);
+
+    unsigned int* ival = (unsigned int*) digested;
```

Comment:
Technically we should do:
```C
unsigned int ival = (digested[0] << 24) | (digested[1] << 16) | (digested[2] << 8) | digested[3];
```
That way, we calculate the same gfid on both little and big endian nodes.  However, if we're currently storing the gfids in the server's metadata, doing this change may corrupt all little endian Unify filesystems out in the wild...

And on a totally unrelated note, we should consider changing the gfid to a int64_t (so it has a larger hash space, and is signed so we can return errors).

Both these issues are outside the scope of this PR though.  Just thought I'd mention them since it jumped out at me while reviewing this code.

Reply:
I'll open a separate issue for this

Reply:
@tonyhutter , better to break things now than later, when the number of active Unify users is small.

Another item to include in the new issue is that we should normalize the path before we hash so that ```/unifyfs/./foo``` is the same as ```/unifyfs/foo```.  Currently, we hash those to two different integers.  Also, we may want to allow for ```chdir("/unifyfs")``` and use of relative paths.

Hashing on the full path makes a ton of sense when we thought support for ```rename()``` would not be needed.  If we do support rename calls, we might also want to think about how we're hashing.  Currently a rename of a top-level directory will require us to rehash every item it its subtree.  That may be fine, but if that's a problem, we'll need to adjust.

Reply:
Issue opened for endianness:
https://github.com/LLNL/UnifyFS/issues/461

Reply:
Decided on today's concall to support 64-bit hash value and Tony's endianess fix from #461 .